cdap-cli/src/main/java/co/cask/cdap/cli/command/metadata/AddMetadataPropertiesCommand.java

```diff
@@ -34,6 +34,10 @@
 
   private final MetadataClient client;
 
+  public static final String ENTITY_DESCRIPTION_STRING = String.format("'<%s>' is of the form " +
+    "'<entity-type>:<namespace-id>.<entity-name>', where '<entity-type>' is one of " +
```

Comment:
But its not just `<program-id>`. I think the best way to handle this is to list out all the options.
Something like: 
```
public static final String ENTITY_DESCRIPTION_STRING = String.format("'<%s>' is of the form '<entity-type>:<entity-id>', where '<entity-type>' is one of " +
    "'artifact', 'app', 'dataset', 'program', 'stream', or 'view', and where '<entity-id>' uniquely identifies a dataset. For datasets and streams, the <entity-id> is composed of namespace and the entity name, such as <namespace-name>.<dataset-name> or <namespace-name>.<stream-name>. Artifacts and apps have an additional 'version' in the <entity-id>, such as <namespace-name>.<artifact-name>.<artifact-version> and <namespace-name>.<app-name>.<app-version>. Stream Views contain the stream that they were created from - <namespace-name>.<stream-name>.<view-name>, while programs contain the application name and the program type - <namespace-name>.<app-name>.<program-type>.<program-name>, where '<program-type>' is one of flow, mapreduce, service, spark, worker, or workflow.", ENTITY);
```

It may be long, but I think its the only way to describe it without any confusion. Perhaps we could use some line break characters (``\n``) to make this easier on the user.

Ultimately though, we shouldn't be using this internal, encoded string. In future, this should change to 

``add metadata tags <comma-separated-tags> to view <view-name> of stream <stream-name>``; OR

``add metadata tags <comma-separated-tags> to <program-type> <program-name> of application <app-name> [version <app-version>]``.

That's more work, but I think its the only way to make this more readable, with autocomplete. Clearly though, its for future.